<a href="https://colab.research.google.com/github/virajvaidya/SRC_ANU_Sampling_1/blob/main/HomeworkW4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import statistics as st
import itertools as it
from matplotlib import pyplot as plt
from scipy.stats import norm

## Q2

In [5]:
population = pd.read_excel('/content/drive/MyDrive/Sampling I - HW4 data.xls')

In [6]:
sample = pd.read_excel('/content/drive/MyDrive/samplehw4.xlsx')

## Q2 a

In [7]:
sample

ID  WEIGHT  BMI  EVERPREG
0   4901     160   26         0
1   3452     142   24         1
2   7961     215   34         1
3   5477     165   27         1
4   1225     120   23         1
5   3329     140   24         1
6   7508     200   33         1
7   6628     183   30         0
8   8457     235   35         1
9   5569     166   30         1
10  5191     160   28         0
11  3677     145   24         0
12  4985     160   25         1
13  6675     185   33         1
14  6168     175   29         1
15  1911     130   22         1
16  4691     155   26         1
17  2222     130   22         1
18  8411     230   38         1
19  7559     200   33         1
20  3482     143   25         0
21  2337     132   24         1
22   998     120   17         1
23  4381     150   25         1
24  2672     135   23         1
25  3787     145   27         1
26  2363     132   27         1
27  1090     120   19         0
28   937     120   19         0
29  8665     246   44         0
30  7820     210   36         1
31  1543     125   24         1
32  6494     180   36         1
33  7114     190   34         1
34   310     108   20         0
35  5608     167   34         1
36  4184     150   27         1
37  2852     137   25         1
38  8513     238   38         0
39  2812     136   25         1

## Q2 b - The sample was collected randomly, so no fixed sampling interval.

# Q2 c - There was no random start. The first observation (ID) selected was totally random with no systematic random start.

## Q2 d : BMI mean is 27.875

In [8]:
bmimean = sample['BMI'].mean()
bmimean

27.875

## Q2 d : BMI SE is 0.95755

## Reason for selecting : Simple Random Sample Assumption and unbiased estimation

In [10]:
bmistandarderror = sample['BMI'].std()/np.sqrt(len(sample))
bmistandarderror

0.9575526362984185

## Q2 e - The design effect is expected to be around 1 since this is a simple random sample.

###  Q1 A sample size of 1547 is required so that the total length of a 95% CI for proprtion of EVERPREG is less than 0.05 (total required sample size) is 1547.

In [12]:
p = 0.6285  # Proportion of EVERPREG = 1 from HOMEWORK 3
Z = 1.96
CI_length = 0.05
DEFF_everpreg = 1.0779  # DEFF for EVERPREG

# Required sample size for SRS
n_srs = (Z**2 * p * (1 - p)) / (CI_length / 2) ** 2

# Adjust for design effect
n_required = n_srs * DEFF_everpreg

# Given proportionate allocation in Homework 3
prop_allocation = np.array([9.97419355, 11.10967742, 9.74193548, 5.52258065, 3.6516129])
prop_allocation_scaled = (prop_allocation / sum(prop_allocation)) * n_required

# Equal allocation (divide total sample size equally among 5 strata)
equal_allocation = np.full(5, n_required / 5)

# Output results
round(n_required), np.round(equal_allocation), np.round(prop_allocation_scaled)



(1547,
 array([309., 309., 309., 309., 309.]),
 array([386., 430., 377., 214., 141.]))

# Total required sample size : 1547
#Equal allocation for each of 5 Strata : 309 each
#Proportionate Allocation:
#S1: 386
#S2: 430
#S3: 377
#S4: 214
#S5: 141

In [13]:
#Neyman allocation weights from HW3
neyman_weights = np.array([5.1091279, 14.55064067, 12.97676182, 4.37404343, 2.98942618])

# Scale Neyman allocation to match required sample size
neyman_allocation_scaled = (neyman_weights / sum(neyman_weights)) * n_required

np.round(neyman_allocation_scaled)


array([198., 563., 502., 169., 116.])

#Neyman Allocation:
#S1 : 198
#S2 : 563
#S3 : 502
#S4 : 169
#S5 : 116

# Q3 a : 3:00